src/printer.rs

```diff
@@ -72,66 +89,95 @@ impl<'a> Printer<'a> {
             self.print_line_number(line_number),
             self.print_git_marker(line_number),
             self.print_line_border(),
-            Some(as_terminal_escaped(
-                &regions,
-                self.options.true_color,
-                self.options.colored_output,
-            )),
+            Some(
+                as_terminal_escaped(
+                    &regions,
+                    self.options.true_color,
+                    self.options.colored_output,
+                ).into(),
+            ),
         ];
 
+        let grid_requested = self.options
+            .output_components
+            .contains(&OutputComponent::Grid);
         write!(
             self.handle,
             "{}",
             decorations
                 .into_iter()
-                .filter_map(|dec| dec)
+                .filter_map(|dec| if grid_requested {
+                    Some(dec.unwrap_or(" ".into()))
```

Comment:
Hmm not sure I follow

Reply:
In a grid we have to at least have a space for a non-existing element to have the right spacing in the left column. Unfortunately I haven't found a cleaner way to express this spacing, without going back to the multiple writes.